### PR TITLE
Add metrics to header

### PR DESCRIPTION
**Pull request by HarmanDhunna**
### Features
- Added metrics to header. Will be used to track the number of requests and responses for each action.

### Other Changes
- Change version semantics to Major.Minor.Patch.PreRelease

### CODE REVIEW

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Release History
 
-## 1.0.0-beta.4 (Unreleased)
+## 1.0.0.4 (Unreleased)
```
Comment (hakimms) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:3]: Can you rebase with main first? There are updates to this file that will be overridden when this gets merged.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md [hunk 2]
```diff
@@ -19,7 +27,7 @@
 - Added createdDateTime to AuthenticationEventContextUser
 - Added new request status type for validation failure.
 - Validation Errors raise 500 response.
-- Added CustomAuthenticaionExtensionId to Data.
+- Added CustomAuthenticationExtensionId to Data.
 - Removed AuthenticationEventsId from Data.
 
 ## 1.0.0-beta.2 (2022-11-08)
```
Comment (hakimms) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:33]: Do we want to update the previous versions to 1.0.0.X and remove -beta?

**Review by hakimms — COMMENTED** [review]
Looks good, just a couple of things to address on the CHANGELOG file

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md (hunk 2 above)
Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:33]: Nah, I just want to correct spelling. lol

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md (hunk 1 above)
Comment (HarmanDhunna) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:3]: I rebased already.

Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:3]: My bad

**Review by hakimms — APPROVED** [review]

**Pull request merged**